lib/joint/MotorJoint.ts

```diff
@@ -0,0 +1,474 @@
+  /** @internal */ m_maxForce: number;
+  /** @internal */ m_maxTorque: number;
+  /** @internal */ m_correctionFactor: number;
+  // Solver temp
+  /** @internal */ m_rA; // Vec2
+  /** @internal */ m_rB; // Vec2
+  /** @internal */ m_localCenterA; // Vec2
+  /** @internal */ m_localCenterB; // Vec2
+  /** @internal */ m_linearError; // Vec2
+  /** @internal */ m_angularError; // float
+  /** @internal */ m_invMassA; // float
+  /** @internal */ m_invMassB; // float
+  /** @internal */ m_invIA; // float
+  /** @internal */ m_invIB; // float
+  /** @internal */ m_linearMass; // Mat22
+  /** @internal */ m_angularMass; // float
+
+  constructor(def: MotorJointDef);
+  constructor(def: MotorJointOpt, bodyA: Body, bodyB: Body);
+  constructor(def: MotorJointDef | MotorJointOpt, bodyA?: Body, bodyB?: Body) {
+    // @ts-ignore
+    if (!(this instanceof MotorJoint)) {
+      return new MotorJoint(def, bodyA, bodyB);
+    }
+
+    def = options(def, DEFAULTS);
+    super(def, bodyA, bodyB);
+    bodyA = this.m_bodyA;
+    bodyB = this.m_bodyB;
+
+    this.m_type = MotorJoint.TYPE;
+
+    this.m_linearOffset = Math.isFinite(def.linearOffset) ? def.linearOffset : bodyA.getLocalPoint(bodyB.getPosition());
+    this.m_angularOffset = Math.isFinite(def.angularOffset) ? def.angularOffset : bodyB.getAngle() - bodyA.getAngle();
+
+    this.m_linearImpulse = Vec2.zero();
+    this.m_angularImpulse = 0.0;
+
+    this.m_maxForce = def.maxForce;
+    this.m_maxTorque = def.maxTorque;
+    this.m_correctionFactor = def.correctionFactor;
+
+    // Solver temp
+    this.m_rA; // Vec2
+    this.m_rB; // Vec2
+    this.m_localCenterA; // Vec2
+    this.m_localCenterB; // Vec2
+    this.m_linearError; // Vec2
+    this.m_angularError; // float
+    this.m_invMassA; // float
+    this.m_invMassB; // float
+    this.m_invIA; // float
+    this.m_invIB; // float
+    this.m_linearMass; // Mat22
+    this.m_angularMass; // float
+
+    // Point-to-point constraint
+    // Cdot = v2 - v1
+    // = v2 + cross(w2, r2) - v1 - cross(w1, r1)
+    // J = [-I -r1_skew I r2_skew ]
+    // Identity used:
+    // w k % (rx i + ry j) = w * (-ry i + rx j)
+
+    // Angle constraint
+    // Cdot = w2 - w1
+    // J = [0 0 -1 0 0 1]
+    // K = invI1 + invI2
+  }
+
+  _serialize() {
+    return {
+      type: this.m_type,
+      bodyA: this.m_bodyA,
+      bodyB: this.m_bodyB,
+      collideConnected: this.m_collideConnected,
+
+      maxForce: this.m_maxForce,
+      maxTorque: this.m_maxTorque,
+      correctionFactor: this.m_correctionFactor,
+
+      linearOffset: this.m_linearOffset,
+      angularOffset: this.m_angularOffset,
+    };
+  };
+
+  static _deserialize(data, world, restore) {
+    data = Object.assign({}, data);
+    data.bodyA = restore(Body, data.bodyA, world);
+    data.bodyB = restore(Body, data.bodyB, world);
+    var joint = new MotorJoint(data);
+    return joint;
+  };
+
+  /**
+   * @internal
+   */
+  _setAnchors(def) {
```

Comment:
I think we can remove this method? Typescrpt says `m_localAnchorA` is not defined and it doesn't seem to be used at least in the class itself